File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/BrandService.java
package com.lsf.pinyougou.sellergoods.service;

import com.lsf.pinyougou.pojo.TbBrand;
import vo.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌列表服务接口
 *
 * @author linshaofeng
 * @date 2020/2/4 13:00
 */
public interface BrandService {

    /**
     * 查询所有品牌
     * @return
     */
    List<TbBrand> findAll();


    /**
     * 品牌分页
     *
     * @param pageNum   当前页码
     * @param pageSize  每页显示的记录数
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件品牌分页查询
     *
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPageLimit(TbBrand brand, int pageNum, int pageSize);


    /**
     * 添加品牌
     *
     * @param brand
     */
    void add(TbBrand brand);


    /**
     * 根据品牌 ID 查询品牌信息
     *
     * @param id
     * @return
     */
    TbBrand findOne(long id);


    /**
     * 修改品牌信息
     * @param brand
     */
    void update(TbBrand brand);


    /**
     * 批量删除品牌
     *
     * @param ids
     */
    void batchDelete(Long[] ids);


    /**
     * 查询所有品牌下拉列表
     *
     * @return
     */
    List<Map> selectBrandList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
// 品牌列表服务模块(自定义服务，依赖 $http 内置服务)
pyg.service('brandService', function ($http) {
    // 查询所有品牌
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };

    // 无条件品牌分页查询
    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    };

    // 多条件品牌分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    };

    // 添加品牌
    this.add = function (brandEntity) {
        return $http.post('../brand/add.do', brandEntity);
    };

    // 修改品牌
    this.update = function (brandEntity) {
        return $http.post('../brand/update.do', brandEntity);
    };

    // 根据品牌 ID 查询品牌信息
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    // 批量删除品牌
    this.batchDelete = function (selectedList) {
        return $http.get('../brand/batchDelete.do?ids=' + selectedList);
    };

    // 查询所有品牌的下拉列表
    this.selectBrandList = function () {
        return $http.get('../brand/selectBrandList.do');
    }
});